workorders.lic

```diff
@@ -332,51 +332,80 @@ class WorkOrders
     end
   end
 
-  def count_combine_rem(stock_room, herb, herb_stock, quantity, herb2)
-    if inside?("first #{herb}", @bag) && inside?("second #{herb}", @bag)
-      bput("get #{herb} from my #{@bag}", 'You get')
-      bput("get #{herb} from my #{@bag}", 'You get')
-      case bput('combine', 'You combine', 'That stack of herbs')
-      when 'That stack of herbs'
-        stow_tool(left_hand)
-        stow_tool(right_hand)
-      when 'You combine'
-        case bput("get #{herb} from my #{@bag}", 'You get', 'What were')
-        when 'You get'
-          case bput('combine', 'You combine', 'That stack of herbs')
-          when 'That stack of herbs', 'You combine'
-            stow_tool(left_hand)
-            stow_tool(right_hand)
-          end
+  # Function which will attempt to combine any stacks of herbs in your inventory, will count them, and determine if more need to be ordered.
+  def count_combine_rem(stock_room, quantity, herb, herb_stock)
+  # Initialize variables
+    found_stack = true;
+    herb_volume_total = 0;
+    last_herb_volume = 0;
+    last_descriptor = "";
+    stack_descriptor = "";
+
+    # We have to only ever use the last word in a multi-word herb.  We will need to be careful when counting because of this.
+    # This is to resolve "red flower" vs "blue flower" because "count third red flower in my bag" fails.
+    # We must instead do a generic "tap third flower in my bag" and determine if we're tapping the right kind of flowers from the response, then actually count them.
+    herb_for_tapping = (herb.gsub(/\s+/m, ' ').strip.split(" ")).last
+
+    # Calculate the volume of the herbs that we will be searching our inventory for and/or buying.
+    # Assume that each unit we need to craft will take 25 volumes of our herb.
+    # Also assume that we also always need double the supply that the workorder would normally take in the event that some of our product is not high enough quality.
```

Comment:
For remedy workorders, the quality any item added to the bun is require to be of superior quality or better.  In my experience, even items that I can normally mastercraft on a regular basis do on occasion end up with imperfections.  This seems to be an unavoidable randomization that happens sometimes.  If you get enough bad RNGs in a row, even simple recipes can sometimes be less than the required quality.  That's why the `;workorders` script checks for verbage when attempting to bun, and will discard any product that wasn't quality enough.  It's WAY overkill to order double of all materials to account for imperfections, but that's easier than trying to go re-buy materials half way through a workorder.